sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -196,8 +196,10 @@ statement
         LIKE pattern=STRING partitionSpec?                             #showTable
     | SHOW TBLPROPERTIES table=tableIdentifier
         ('(' key=tablePropertyKey ')')?                                #showTblProperties
-    | SHOW COLUMNS (FROM | IN) tableIdentifier
-        ((FROM | IN) db=errorCapturingIdentifier)?                     #showColumns
+    | SHOW COLUMNS (FROM | IN)
+        ((tableName=errorCapturingIdentifier
+          (FROM | IN) namespace=multipartIdentifier) |
+        table=multipartIdentifier)                                     #showColumns
```

Comment:
So this works, but makes the syntax rule a little bit complicated. Another approach is
```
SHOW COLUMNS (FROM | IN) tableName= multipartIdentifier
    ((FROM | IN) namespace=multipartIdentifier)?
```

In `visitShowColumns`, if `namespace` is specified, we check the length of `tableName`, and throw exception if it has more than one part.

By doing this, we also have a chance to give better error message, telling users that they can only use single-part table name if `FROM namespace` is specified.

Reply:
Ok, I was not sure about the error message, it was not very intuitive. It is better as the way you say me.

Reply:
@cloud-fan 
I reviewed the test cases and code and I arrived the conclusion that something like that is permitted:
SHOW COLUMNS FROM db1.t1 IN db1
So I have change this pull request to allow it and don't break tests in other parts.
Are you agree with it?
Thanks